xarray/core/duck_array_ops.py

```diff
@@ -149,6 +149,33 @@ def trapz(y, x, axis):
 )
 
 
+def astype(data, **kwargs):
+    try:
+        return data.astype(**kwargs)
+    except TypeError as e:
```

Comment:
would it be better to do something like
```python
        try:
            import sparse
        except ImportError:
            sparse = None

        if (
            sparse is not None
            and isinstance(data, sparse.C00)
            and LooseVersion(sparse.__version__) < LooseVersion("0.10.0")
            and "casting" in kwargs
        ):
            # warn
            kwargs.pop("casting")
        return data.astype(**kwargs)
```
that way, we wouldn't call `data.astype(**kwargs)` twice